ledfx/api/websocket.py

```diff
@@ -297,6 +300,36 @@ def audio_stream_data_handler(self, message):
             message.get("data").values(), dtype=np.float32
         )
 
+    @websocket_handler("audio_stream_data_v2")
+    def audio_stream_data_base64_handler(self, message):
+        # Max value for signed 16-bit values.
+        MAX_VAL = 32767
+        if not ACTIVE_AUDIO_STREAM:
+            return
+
+        client = message.get("client")
+
+        if ACTIVE_AUDIO_STREAM.client != client:
+            return
+        try:
+            decoded = pybase64.b64decode(message.get("data"))
```

Comment:
Copilot suggests putting this directly into a numpy array and adding some error handling

```decoded = np.frombuffer(base64.b64decode(message.get("data")), dtype=np.int16)
        if len(decoded) % 2 != 0:
            raise ValueError("Incorrect data length after decoding.")
```

Probably want some logging and stuff but don't want to think too hard if you've benched this already and its slower etc.

Reply:
Just noticed that you're suggesting frombuffer here, as mentioned in another comment, this reduces performance.